packages/compass-import-export/src/modules/import.ts

```diff
@@ -268,142 +309,282 @@ export const startImport = () => {
         transform,
       }
     );
-    const source = fs.createReadStream(fileName, 'utf8');
 
-    const stripBOM = stripBomStream();
+    const fields: Record<string, CSVParsableFieldType> = {};
+    for (const [name, type] of transform) {
+      if (exclude.includes(name)) {
+        continue;
+      }
+      fields[name] = typeToCSVParsableFieldType(type || 'mixed');
+    }
 
-    const parser = createParser({
-      fileName,
-      fileType,
-      delimiter,
-      fileIsMultilineJSON,
-    });
+    const source = fs.createReadStream(fileName, 'utf8');
 
-    const applyTypes = transformProjectedTypesStream({
-      exclude,
-      transform,
-      ignoreBlanks,
-    });
+    if (useNewImportBackend) {
+      let promise;
 
-    const dest = createCollectionWriteStream(dataService!, ns, stopOnErrors);
+      // TODO: log file, but probably only useful once we have the toast
+      //const logPath = path.join(app.getPath('logs'), path.basename(fileName) + '.log');
+      //const output = fs.createWriteStream(logPath);
 
-    dest.on('progress', (stats: CollectionStreamProgress) => {
-      dispatch(onProgress(stats));
-    });
+      // TODO: abort controller, not source and dest
+      dispatch({ type: STARTED });
 
-    const progress = createProgressStream(
-      fileSize,
-      function (
-        err: Error | undefined,
-        info: {
-          percentage: number;
-          transferred: number;
-          length: number;
-          remaining: number;
-          eta: number;
-          runtime: number;
-          delta: number;
-          speed: number;
-        }
-      ) {
-        if (err) return;
-        dispatch(onGuesstimatedProgress(info.transferred, info.length));
-      }
-    );
+      const errors: ErrorJSON[] = [];
+      const errorCallback = (err: ErrorJSON) => {
+        errors.push(err);
+      };
 
-    const importSizeGuesstimator = createImportSizeGuesstimator(
-      source,
-      fileSize,
-      function (err: Error | undefined, guesstimatedTotalDocs: number) {
-        if (err) return;
-        progress.setLength(guesstimatedTotalDocs);
-        dispatch(onGuesstimatedDocsTotal(guesstimatedTotalDocs));
+      const progressCallback = _.throttle(function (
+        numDocs: number,
+        numBytes: number
+      ) {
+        console.log('progress', numDocs, errors.length);
+
+        // for now, call onGuesstimatedDocsTotal() so that the existing progress bar worlks
+        const guessedTotal = Math.max(numDocs, Math.ceil(fileSize / numBytes));
+        dispatch(onGuesstimatedDocsTotal(guessedTotal));
+
+        dispatch(onGuesstimatedProgress(numDocs, guessedTotal));
+
+        dispatch(
+          onProgress({
+            docsWritten: numDocs, // TODO: this is a lie
```

Comment:
I don't really see why we even need docsWritten. Progress should just be one thing and I think docsProcessed is fine?

If we DO need both processed and written, then I think we should add a consistent third param to the progress callback in importJSON and importCSV.